src/main/java/org/jenkinsci/plugins/github_branch_source/DefaultGitHubNotificationStrategy.java

```diff
@@ -8,16 +8,21 @@
 import org.jenkinsci.plugins.displayurlapi.DisplayURLProvider;
 import org.kohsuke.github.GHCommitState;
 
-public class GitHubBuildStatusNotificationStrategy {
-    protected Job<?, ?> job;
-    protected Run<?, ?> build;
-    protected TaskListener listener;
-    protected SCMSource source;
-    protected SCMHead head;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
 
-    public GitHubBuildStatusNotificationStrategy() {}
+/**
+ * @since TODO
+ */
+public class DefaultGitHubNotificationStrategy extends AbstractGitHubNotificationStrategy {
+    public DefaultGitHubNotificationStrategy() {}
 
-    protected String generateContext() {
+    /**
+     * @since TODO
+     */
+    protected String generateContext(GitHubNotificationContext notificationContext, TaskListener listener) {
```

Comment:
would it make more sense to move this to a `getDefaultContext()` method on `GitHubNotificationContext`?

Reply:
Do you mean leave the method, but move the logic into a new method in `GitHubNotificationContext`?

This is the approach I'm using on the trait implementation end for testing
https://gist.github.com/steven-foster/ea9e2067d2fa4aa930e8ad521b886842

Reply:
Your call, but if we move it then people do not need to inherit from `DefaultGitHubNotificationStrategy`